docs/advanced/testing.md

```diff
@@ -61,3 +61,115 @@ ResiliencePipelineDescriptor descriptor = pipeline.GetPipelineDescriptor();
+await sut.RetrieveData();
+```
+<!-- endSnippet -->
+
+**Reasoning**:
+
+- Depending on the your strategy setup, the execution time might be greatly increased (from milliseconds to seconds).
+  - For example if you have a retry with 3 `MaxRetryAttempts` and 1 seconds `Delay`.
+- This violates one of the F.I.R.S.T testing principles, namely the **Fast** one.
+
+✅ DO
+
+Use `Empty` or mock `ExecuteAsync`:
+
+- If you want to test your to-be-decorated code like there is no strategy applied:
+
+<!-- snippet: testing-pattern-1-1 -->
+```cs
+// Arrange
+var timeoutMock = ResiliencePipeline<HttpResponseMessage>.Empty;
+var downstreamMock = Substitute.For<IDownstream>();
+// setup the to-be-decorated method in a way that suits for your test case
+
+// Act
+var sut = new SUT(downstreamMock, timeoutMock);
+await sut.RetrieveData();
+```
+<!-- endSnippet -->
+
+- If you want to verify that your code handles success cases as expected:
+
+<!-- snippet: testing-pattern-1-2 -->
+```cs
+// Arrange
+var downstreamMock = Substitute.For<IDownstream>();
+
+var timeoutMock = Substitute.For<MockableResilienceStrategy>();
+timeoutMock.ExecuteAsync(Arg.Any<Func<CancellationToken, ValueTask<HttpResponseMessage>>>(), Arg.Any<CancellationToken>())
+        .Returns(new HttpResponseMessage(HttpStatusCode.Accepted));
+
+var optionsMock = Substitute.For<ResilienceStrategyOptions>();
+
+var pipeline = new ResiliencePipelineBuilder<HttpResponseMessage>()
+    .AddStrategy(_ => timeoutMock, optionsMock)
+    .Build();
+
+// Act
+var sut = new SUT(downstreamMock, pipeline);
+await sut.RetrieveData();
+```
+<!-- endSnippet -->
+
+- If you want to verify that your code handles failure cases as expected:
+
+<!-- snippet: testing-pattern-1-3 -->
+```cs
+// Arrange
+var downstreamMock = Substitute.For<IDownstream>();
+
+var timeoutMock = Substitute.For<MockableResilienceStrategy>();
+timeoutMock.ExecuteAsync(Arg.Any<Func<CancellationToken, ValueTask<HttpResponseMessage>>>(), Arg.Any<CancellationToken>())
+        .ThrowsAsync(new TimeoutRejectedException());
+
+var optionsMock = Substitute.For<ResilienceStrategyOptions>();
+
+var pipeline = new ResiliencePipelineBuilder<HttpResponseMessage>()
+    .AddStrategy(_ => timeoutMock, optionsMock)
+    .Build();
+
+// Act
+var sut = new SUT(downstreamMock, pipeline);
+await sut.RetrieveData();
+```
+<!-- endSnippet -->
+
+**Reasoning**:
+
+- Polly currently does not support direct mocking of `ResiliencePipeline`.
+  - As a workaround you can derive from the `ResilienceStrategy` class.
+  - You can expose an `ExecuteAsync` method with the same signature as you it inside the system-under-test.
+  - Then you add that strategy to a pipeline.
+- This approach allows you to define the expected behaviour for your resilience pipeline which meets your test case's needs.
+
+> [!NOTE]
+> The above presented solution is a workaround and later Polly might add support for this use case as well in the `Polly.Testing` package.
```

Comment:
I'm not sure we should have this here in our docs without a concrete issue somewhere asking for it. Unless you really really want to advocate for such a thing, I would just remove this note.